autotest/gdrivers/mem.py

```diff
@@ -175,6 +175,35 @@ def test_mem_2():
 
         dsup = None
 
+    wktll = """GEOGCS[\\"WGS 84\\",DATUM[\\"WGS_1984\\",SPHEROID[\\"WGS 84\\",6378137,298.257223563,AUTHORITY[\\"EPSG\\",\\"7030\\"]],AUTHORITY[\\"EPSG\\",\\"6326\\"]],PRIMEM[\\"Greenwich\\",0,AUTHORITY[\\"EPSG\\",\\"8901\\"]],UNIT[\\"degree\\",0.0174532925199433,AUTHORITY[\\"EPSG\\",\\"9122\\"]],AXIS[\\"Latitude\\",NORTH],AXIS[\\"Longitude\\",EAST],AUTHORITY[\\"EPSG\\",\\"4326\\"]]"""
+
+    dsnames2 = [
+        "MEM:::DATAPOINTER=0x%X,SPATIALREFERENCE=+proj=laea +lon_0=147 +lat_0=-42,GEOTRANSFORM=-1e+06/1953.125/0/1e+06/0/-3906.25,PIXELS=%d,LINES=%d,DATATYPE=Float32"
+        % (p, width, height),
+        "MEM:::DATAPOINTER=0x%X,SPATIALREFERENCE=bogus,GEOTRANSFORM=-1e+06/1953.125/0/1e+06/0/-3906.25,PIXELS=%d,LINES=%d,DATATYPE=Float32"
+        % (p, width, height),
+        'MEM:::DATAPOINTER=0x%X,SPATIALREFERENCE="%s",GEOTRANSFORM=-1e+06/1953.125/0/1e+06/0/-3906.25,PIXELS=%d,LINES=%d,DATATYPE=Float32'
+        % (p, wktll, width, height),
+    ]
+    print(dsnames2[2])
+    cnt = 0
+    for dsname in dsnames2:
+        cnt = cnt + 1
+        for i in range(width * height):
+            float_p[i] = 5.0
+
+        dsro = gdal.Open(dsname)
+        if dsro is None:
+            free(p)
+            pytest.fail("opening MEM dataset failed in read only mode.")
+        assert dsro.GetGeoTransform() == (-1e06, 1953.125, 0, 1e06, 0, -3906.25)
+        if cnt == 1:
```

Comment:
Can you add the expected projection ref in the dsnames2 array (make it an array of tuples) ? 

It will be easier to read and to maintain.

Reply:
like this? 

```python
sref = ("+proj=laea +lon_0=147 +lat_0=-42", 
        "bogus",
        """GEOGCS[\\"WGS 84\\",DATUM[\\"WGS_1984\\",SPHEROID[\\"WGS 84\\",6378137,298.257223563,AUTHORITY[\\"EPSG\\",\\"7030\\"]],AUTHORITY[\\"EPSG\\",\\"6326\\"]],PRIMEM[\\"Greenwich\\",0,AUTHORITY[\\"EPSG\\",\\"8901\\"]],UNIT[\\"degree\\",0.0174532925199433,AUTHORITY[\\"EPSG\\",\\"9122\\"]],AXIS[\\"Latitude\\",NORTH],AXIS[\\"Longitude\\",EAST],AUTHORITY[\\"EPSG\\",\\"4326\\"]]""")
        
dsnames2 = [
        "MEM:::DATAPOINTER=0x%X,SPATIALREFERENCE=%s,GEOTRANSFORM=-1e+06/1953.125/0/1e+06/0/-3906.25,PIXELS=%d,LINES=%d,DATATYPE=Float32"
        % (p, sref[0], width, height),
        "MEM:::DATAPOINTER=0x%X,SPATIALREFERENCE=%s,GEOTRANSFORM=-1e+06/1953.125/0/1e+06/0/-3906.25,PIXELS=%d,LINES=%d,DATATYPE=Float32"
        % (p, sref[1], width, height),
        'MEM:::DATAPOINTER=0x%X,SPATIALREFERENCE="%s",GEOTRANSFORM=-1e+06/1953.125/0/1e+06/0/-3906.25,PIXELS=%d,LINES=%d,DATATYPE=Float32'
        % (p, sref[2], width, height),
    ]

```

apologies I'm very unfamiliar with python

Reply:
> like this?
> 
> ```python
> sref = ("+proj=laea +lon_0=147 +lat_0=-42", 
>         "bogus",
>         """GEOGCS[\\"WGS 84\\",DATUM[\\"WGS_1984\\",SPHEROID[\\"WGS 84\\",6378137,298.257223563,AUTHORITY[\\"EPSG\\",\\"7030\\"]],AUTHORITY[\\"EPSG\\",\\"6326\\"]],PRIMEM[\\"Greenwich\\",0,AUTHORITY[\\"EPSG\\",\\"8901\\"]],UNIT[\\"degree\\",0.0174532925199433,AUTHORITY[\\"EPSG\\",\\"9122\\"]],AXIS[\\"Latitude\\",NORTH],AXIS[\\"Longitude\\",EAST],AUTHORITY[\\"EPSG\\",\\"4326\\"]]""")
>         
> dsnames2 = [
>         "MEM:::DATAPOINTER=0x%X,SPATIALREFERENCE=%s,GEOTRANSFORM=-1e+06/1953.125/0/1e+06/0/-3906.25,PIXELS=%d,LINES=%d,DATATYPE=Float32"
>         % (p, sref[0], width, height),
>         "MEM:::DATAPOINTER=0x%X,SPATIALREFERENCE=%s,GEOTRANSFORM=-1e+06/1953.125/0/1e+06/0/-3906.25,PIXELS=%d,LINES=%d,DATATYPE=Float32"
>         % (p, sref[1], width, height),
>         'MEM:::DATAPOINTER=0x%X,SPATIALREFERENCE="%s",GEOTRANSFORM=-1e+06/1953.125/0/1e+06/0/-3906.25,PIXELS=%d,LINES=%d,DATATYPE=Float32'
>         % (p, sref[2], width, height),
>     ]
> ```
> 
> apologies I'm very unfamiliar with python

No need to apologize!

Actually I was thinking at something like:

```python
test_data = [
       ( "MEM:::DATAPOINTER=0x%X,SPATIALREFERENCE=+proj=laea +lon_0=147 +lat_0=-42,GEOTRANSFORM=-1e+06/1953.125/0/1e+06/0/-3906.25,PIXELS=%d,LINES=%d,DATATYPE=Float32"
        % (p, width, height), "+proj=laea +lon_0=147 +lat_0=-42"),
       #  all other cases
    ]

# remove the print instruction and `cnt`

for ds_definition, expected_sr in test_data:
    # run your test here

Reply:
oh wow, yes right I think in R and this is very different, thanks will do 🙏

Reply:
@mdsumner @elpaso See Sean's remark at https://lists.osgeo.org/pipermail/gdal-dev/2023-February/056928.html
I was also thinking the same, but as this was an extension of an existing test that used loops, I didn't insist